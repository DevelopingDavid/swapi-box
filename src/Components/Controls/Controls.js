import React, { Component } from 'react';

export default class Controls extends Component{
  constructor(props) {
    super(props);
  }

  recievePeople =  async () => {
    let peopleArray = [];
    if(this.props.people.length === 0) {
      const response = await fetch('https://swapi.co/api/people/?page=1')
      const people = await response.json();
      peopleArray.push(...people.results);
      const homeworld = await this.recieveHomeWorld(peopleArray);
      const finalData = await this.recieveSpecies(homeworld);
      this.props.retrievePeople(finalData);
    }
  }

  recieveHomeWorld = (people) => {
    const unresolvedPromises = people.map(async person => {
      const response = await fetch(person.homeworld);
      const home = await response.json();
      return ({...person, homeworld: home.name, population: home.population});
    });
    return Promise.all(unresolvedPromises);
  }

  recieveSpecies = (people) => {
    const unresolvedPromises = people.map(async person => {
      const response = await fetch(person.species[0]);
      const species = await response.json();
      return ({name: person.name, homeworld: person.homeworld, population: person.population, species: species.name, language: species.language});
    });
    return Promise.all(unresolvedPromises);
  }
  
  render() {
    return (
      <section>
        <button onClick={this.recievePeople}>People</button>
        <button>Planets</button>
        <button>Vehicles</button>
        <button>Favorites</button>
        <h3>0</h3>
      </section>
    )
  }
} 