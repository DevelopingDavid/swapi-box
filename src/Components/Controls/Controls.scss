button {
  width: 100px;
  height: 30px;
  border-radius: 10px;
  margin: 0 10px;
  border: 2px solid #feda4a;
  background: transparent;
  color: white;
  font-family: sans-serif;
  font-weight: 700;
  font-size: 1rem;
  outline: none;
  &:hover {
    background: #feda4a;
    color: black;
  }
  &:focus {
    background: #feda4a;
    color: black;
  }
}