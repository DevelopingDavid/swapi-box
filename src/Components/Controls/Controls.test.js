import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Controls/>
    )
  });

  it('should create snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});