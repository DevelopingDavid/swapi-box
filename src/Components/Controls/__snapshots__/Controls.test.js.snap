// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should create snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          People
        </button>,
        <button>
          Planets
        </button>,
        <button>
          Vehicles
        </button>,
        <button>
          Favorites
        </button>,
        <h3>
          0
        </h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "0",
        },
        "ref": null,
        "rendered": "0",
        "type": "h3",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            People
          </button>,
          <button>
            Planets
          </button>,
          <button>
            Vehicles
          </button>,
          <button>
            Favorites
          </button>,
          <h3>
            0
          </h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "0",
          },
          "ref": null,
          "rendered": "0",
          "type": "h3",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
